File: java/src/main/behavioral/chainofresponsibility/ConcreteHandlerA.java
package behavioral.chainofresponsibility;

// ConcreteHandlerA: Handles certain requests and passes the rest
public class ConcreteHandlerA extends Handler {
    @Override
    public void handleRequest(String request) {
        if (request.equalsIgnoreCase("A")) {
            System.out.println("ConcreteHandlerA handled request: " + request);
        } else if (nextHandler != null) {
            nextHandler.handleRequest(request);
        } else {
            System.out.println("No handler could process request: " + request);
        }
    }
}

File: python/src/behavioral/chain_of_responsibility/concrete_handler_a.py
from .abstract_handler import AbstractHandler


class ConcreteHandlerA(AbstractHandler):
    """
    Handles requests starting with 'A'.
    """

    def handle(self, request: str) -> str:
        if request.startswith("A"):
            return f"Handler A processed: {request}"
        return super().handle(request)
